### ISSUE TITLE
[Docs] mark function usage - [TypeInferenceError] Inferred elem type differs from existing elem type: (7) vs (1)

### ISSUE THREAD

**Issue by zjhthu** [open-mmlab/mmdeploy#2357]
### 📚 The doc issue

I met problems when converting the RTM-instance segmentation model to the MNN model. So, I want to partition the model for debugging.  The post-processing function is split at this [line](https://github.com/open-mmlab/mmdeploy/blob/main/mmdeploy/codebase/mmdet/models/dense_heads/rtmdet_ins_head.py#L177) as following. But I got an error when converting the model to onnx: `[TypeInferenceError] Inferred elem type differs from existing elem type: (7) vs (1)

```
@mark('rtmdet_ins_nms_post_process', 
    inputs=['keep_top_k', 'dets', 'labels', 'kernels', 'priors', 'mask_feats'],
    outputs=['dets', 'labels', 'masks'])
def _nms_post_process_with_mask_static(self, keep_top_k, dets, labels, kernels, priors, mask_feats):
    # topk or sort
    is_use_topk = keep_top_k > 0 and \
        (torch.onnx.is_in_onnx_export() or keep_top_k < dets.shape[1])
    if is_use_topk:
        _, topk_inds = dets[:, :, -1].topk(keep_top_k, dim=1)
    else:
        _, topk_inds = dets[:, :, -1].sort(dim=1, descending=True)
    topk_inds = topk_inds.squeeze(0)
    dets = dets[:, topk_inds, ...]
    labels = labels[:, topk_inds, ...]
    kernels = kernels[:, topk_inds, ...]
    priors = priors[topk_inds, ...]
    mask_logits = _mask_predict_by_feat_single(self, mask_feats, kernels[0],
                                               priors)
    stride = self.prior_generator.strides[0][0]
    mask_logits = F.interpolate(
        mask_logits.unsqueeze(0), scale_factor=stride, mode='bilinear')
    masks = mask_logits.sigmoid()
    return dets, labels, masks
```

The config for deployment is attached below:
```
_base_ = [
    '../_base_/base_instance-seg_static.py',
    '../../_base_/backends/onnxruntime.py'
]

# Notice: Do not set input_shape in onnx_config!
# This will result in an incorrect scale_factor!
# The input shape will be automatically inferred
# from the model's test_pipeline config.
onnx_config = dict(input_shape=None)
codebase_config = dict(post_processing=dict(export_postprocess_mask=True))
partition_config = dict(
    type='rtmdet-ins_partition', # the partition policy name
    apply_marks=True, # should always be set to True
    partition_cfg=[
        dict(
            save_file='rtmdet-ins.onnx', # filename to save the partitioned onnx model
            start=['detector_forward:input'], # [mark_name:input/output, ...]
            end=['rtmdet_ins_nms_post_process:input'],
            output_names=['keep_top_k', 'dets', 'labels', 'kernels', 'priors', 'mask_feats']
        )
    ])
```

Does anyone have suggestions?

**Comment by RunningLeon** [comment]
@zjhthu hi, you can directly return 'dets', 'labels', 'kernels', 'priors', 'mask_feats' and change the `output_names` in deploy config instead of using mark.
